emcc.py

```diff
@@ -1913,6 +1913,15 @@ def get_full_import_name(name):
           suggestion = 'decrease MAXIMUM_MEMORY'
         exit_with_error('emmalloc only works on <2GB of memory. Use the default allocator, or ' + suggestion)
 
+    # various settings require JS post-processing
+    if shared.Settings.OPT_LEVEL >= 2 or \
+       shared.Settings.USE_CLOSURE_COMPILER or \
+       shared.Settings.CAN_ADDRESS_2GB or \
+       shared.Settings.USE_ASAN or \
+       shared.Settings.SAFE_HEAP or \
+       (shared.Settings.USE_PTHREADS and shared.Settings.ALLOW_MEMORY_GROWTH):
```

Comment:
Could we make runtime EM_ASM replacement work also for multithreaded programs?

Reply:
@pfaffe Multithreaded yes, but this handles the case of multithreading + growth. That case requires JS rewriting (to constantly check for growth). There is spec work to allow us to avoid that, by @syg (@syg, incidentally this shows another benefit of that proposal, that it would simplify the toolchain side of things).